### DISCUSSION TITLE
Updated OS X Instructions

### DISCUSSION THREAD

**Discussion by AllysonOfficial** [debauchee/barrier#1494] [General]
Steps that worked for me on 3 different Mac OS Machines - Big Sur, Mojave, and Big Sur (Macbook Pro).  

1. Install Barrier by downloading the DMG file from the releases and install by Copying the application to your Applications Folder on the Mac. (Do this on all machines that you wish to have the application running)
   a. Please note, one machine needs to be the server and the rest will be set up as the client machines to connect to the server.
2. Due to the fact that SSL is not installed during the installation of the application, you will need to create this manually using this: 
   a.  **openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem**
   b.  THIS NEEDS TO BE DONE ON ALL MACHINES. (Since each machine needs to have it's own unique SSL SHA1 RSA fingerprint, the token creation of this command on each machine is what worked for me.  I do believe that you can copy the PEM from the server to each of the clients and that will work too, but knowing that each machine works most stable using their own SHA1 RSA fingerprint, i created all unique ones to ensure no reciprocity)
3.  Once you have created a new SSL on all the machines which are going to be running the Barrier App, you then need to determine which machine is going to be the Server and which will be the client. 
4. After you have done step 2 and moved thru the step, shut down the barrier application and restart it on all machines.  IT will show that SSL is enabled with the fingerprint listed at the bottom.  
5. On the Server machine you will need to create the configuration file and save it.
   a. Find out what the clients' hostname is by going into the System Preferences, and then Sharing, and under Computer Name, you will see the name - mac01.local (name for example purposes)
  b.  This is the SCREEN NAME to be used in the configuration settings on the server. 
  c. On the server - mac-srv.local (name for example purposes) launch the barrier application and the settings window will appear and you will see the SHA1 SSL displayed below and the option for "Configure the Server Interactively" checkbox, and "Load from Configuration File" checkbox.  Leave it in the 1st one for now.  
  d.  Click the "Configure Server" button on the page. 
  e.  The next screen will show the hostname / screenname of the server mac-srv.local (example name) and a trash can in the left upper corner and Plus in the upper right, Click the Plus
  f.  This will allow you to add a client screen name to the configuration. 
  g.  On the next window, type in the hostname from the client you noted above, which is the mac01.local , and in alias, type in mac01.
  h.  If you wish to edit any of the other settings you can, but not necessary.
  i.   Once you have done, this, click OK and the new Computer will show next to the server screen name in the window.  Press OK again.  
  j. **PLEASE NOTE: MAKE SURE BARRIER IS STOPPED WHILE DOING THIS** Makes things so much easier. 
  k.  Now that you have your configuration set, now save it by Clicking 'Barrier' in the very upper left of the Menu Bar of the Mac, and last option - Save Configuration.  Select a local location in your /Users/<user>/Documents/barrier.conf as the file name. 
6.  After you have done that, check the box "Load from configuration File" and browse for the file you just created "barrier.conf" in the /Users/<user>/Documents/barrier.conf
7. Click Start at the bottom and it should start and show RUNNING as the status. 
8. On the client machine, assuming you have created the SSL as above, restarted the application, and now you can click Start on the client and it will establish the connection and be working.  
9. **You can open the Log file while its running by right clicking the icon in the toolbar at the top and selecting Show Log to see if there are any errors** 

10.  Now it should be working and you should be good to go.  
Please note if you don't save the config file, and then load the file to start the server and the service, you will get the following failure error: **unrecognised client name "xxx", check server config**

Thanks again, and if you have any questions, please feel free to message me.

**Comment by AllysonOfficial** [comment]
Please Note also that if you lock either machine screens, the server only hosts the keyboard input, and you will have to unlock that machine in order for it to be 2 way <--- ---->.  ;-)

**Comment by Sandrat954** [comment]
Anytime I try to install Barrier to my Macbook Pro, I get this message in the image. 


<img width="1199" alt="Screenshot 2024-06-15 at 2 35 50 AM" src="https://github.com/debauchee/barrier/assets/172893464/bdcc3c0e-f5d8-449b-a40f-7903953f9926">

**Reply by antoine-truong** [reply]
Open the Applications folder with Finder.
First Drag and Drop Barrier in the Application folder.
Then, from the Application folder, hold Control and click on the Barrier application, then click Open

**Comment by mkv111** [comment]
This worked for me following first 3 steps.

Windows 10 (Server) -> Mac OS Sonoma 14.5
SSL was created on Windows and as soon as it was attached to the server application, Client detected by itself and asked for a confirmation to acquire the SSL certificate request.

Also, I found a missed step to move the Barrier.pem file to the installed/required location "C:\Users\<USERNAME>\AppData\Local\Barrier\SSL\".


> Steps that worked for me on 3 different Mac OS Machines - Big Sur, Mojave, and Big Sur (Macbook Pro).
> 
> 1. Install Barrier by downloading the DMG file from the releases and install by Copying the application to your Applications Folder on the Mac. (Do this on all machines that you wish to have the application running)
>    a. Please note, one machine needs to be the server and the rest will be set up as the client machines to connect to the server.
> 2. Due to the fact that SSL is not installed during the installation of the application, you will need to create this manually using this:
>    a.  **openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem**
>    b.  THIS NEEDS TO BE DONE ON ALL MACHINES. (Since each machine needs to have it's own unique SSL SHA1 RSA fingerprint, the token creation of this command on each machine is what worked for me.  I do believe that you can copy the PEM from the server to each of the clients and that will work too, but knowing that each machine works most stable using their own SHA1 RSA fingerprint, i created all unique ones to ensure no reciprocity)
> 3. Once you have created a new SSL on all the machines which are going to be running the Barrier App, you then need to determine which machine is going to be the Server and which will be the client.
> 4. After you have done step 2 and moved thru the step, shut down the barrier application and restart it on all machines.  IT will show that SSL is enabled with the fingerprint listed at the bottom.
> 5. On the Server machine you will need to create the configuration file and save it.
>    a. Find out what the clients' hostname is by going into the System Preferences, and then Sharing, and under Computer Name, you will see the name - mac01.local (name for example purposes)
>    b.  This is the SCREEN NAME to be used in the configuration settings on the server.
>    c. On the server - mac-srv.local (name for example purposes) launch the barrier application and the settings window will appear and you will see the SHA1 SSL displayed below and the option for "Configure the Server Interactively" checkbox, and "Load from Configuration File" checkbox.  Leave it in the 1st one for now.
>    d.  Click the "Configure Server" button on the page.
>    e.  The next screen will show the hostname / screenname of the server mac-srv.local (example name) and a trash can in the left upper corner and Plus in the upper right, Click the Plus
>    f.  This will allow you to add a client screen name to the configuration.
>    g.  On the next window, type in the hostname from the client you noted above, which is the mac01.local , and in alias, type in mac01.
>    h.  If you wish to edit any of the other settings you can, but not necessary.
>    i.   Once you have done, this, click OK and the new Computer will show next to the server screen name in the window.  Press OK again.
>    j. **PLEASE NOTE: MAKE SURE BARRIER IS STOPPED WHILE DOING THIS** Makes things so much easier.
>    k.  Now that you have your configuration set, now save it by Clicking 'Barrier' in the very upper left of the Menu Bar of the Mac, and last option - Save Configuration.  Select a local location in your /Users//Documents/barrier.conf as the file name.
> 6. After you have done that, check the box "Load from configuration File" and browse for the file you just created "barrier.conf" in the /Users//Documents/barrier.conf
> 7. Click Start at the bottom and it should start and show RUNNING as the status.
> 8. On the client machine, assuming you have created the SSL as above, restarted the application, and now you can click Start on the client and it will establish the connection and be working.
> 9. **You can open the Log file while its running by right clicking the icon in the toolbar at the top and selecting Show Log to see if there are any errors**
> 10. Now it should be working and you should be good to go.
>     Please note if you don't save the config file, and then load the file to start the server and the service, you will get the following failure error: **unrecognised client name "xxx", check server config**
> 
> Thanks again, and if you have any questions, please feel free to message me.

**Reply by gajananbhat** [reply]
Can confirm. This worked for me.

> This worked for me following first 3 steps.
> 
> Windows 10 (Server) -> Mac OS Sonoma 14.5 SSL was created on Windows and as soon as it was attached to the server application, Client detected by itself and asked for a confirmation to acquire the SSL certificate request.
> 
> Also, I found a missed step to move the Barrier.pem file to the installed/required location "C:\Users<USERNAME>\AppData\Local\Barrier\SSL".
>
